trieste/models/gpflux/models.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import warnings
+
+import tensorflow as tf
+from gpflux.layers import GPLayer
+from gpflux.models import DeepGP
+from gpflux.models.deep_gp import sample_dgp
+
+from ...data import Dataset
+from ...types import TensorType
+from ...utils import jit
+from ..interfaces import TrainableProbabilisticModel
+from ..optimizer import Optimizer, TFOptimizer
+from .interface import GPfluxPredictor
+
+
+class DeepGaussianProcess(GPfluxPredictor, TrainableProbabilisticModel):
+    """
+    A :class:`TrainableProbabilisticModel` wrapper for a GPflux :class:`~gpflow.models.DeepGP` with
+    only standard :class:`GPLayer`: this class does not support keras layers, latent variable
+    layers, etc.
+    """
+
+    def __init__(self, model: DeepGP, optimizer: Optimizer | None = None):
```

Comment:
as in interface, I would pass `tf.optimizers.Optimizer` here and arguments object for `fit` method

Reply:
This isn't entirely possible this way because of what `create_model_interface` in `ModelConfig` from `models\config.py` expects you to pass to it. I guess one solution would be to write a new `create_model_interface` method for the `GPfluxModelConfig` but that might overcomplicate things?